packages/boot/test/bootstrapTests/test-addAssets.ts

```diff
@@ -0,0 +1,196 @@
+  const {
+    advanceTimeTo,
+    readLatest,
+    // @ts-expect-error override
+    proposal,
+  } = t.context;
+
+  // stringify, modify and parse because modifying a deep copy was fragile.
+  const proposalJSON = JSON.stringify(proposal);
+  const proposalMod = proposalJSON.replaceAll('STARS', 'COMETS');
+  const proposalNew = JSON.parse(proposalMod);
+
+  const bridgeMessage = {
+    type: 'CORE_EVAL',
+    evals: proposalNew.evals,
+  };
+
+  const { EV } = t.context.runUtils;
+
+  const auctioneerKit = await EV.vat('bootstrap').consumeItem('auctioneerKit');
+  const schedules = await EV(auctioneerKit.creatorFacet).getSchedule();
+  const { liveAuctionSchedule, nextAuctionSchedule } = schedules;
+  const nextEndTime = liveAuctionSchedule
+    ? liveAuctionSchedule.endTime
+    : nextAuctionSchedule.endTime;
+  const fiveMinutes = harden({
+    relValue: 5n * 60n,
+    timerBrand: nextEndTime.timerBrand,
+  });
+  const nextQuiescentTime = TimeMath.addAbsRel(nextEndTime, fiveMinutes);
+  await advanceTimeTo(nextQuiescentTime);
+
+  const coreEvalBridgeHandler = await EV.vat('bootstrap').consumeItem(
+    'coreEvalBridgeHandler',
+  );
+  await EV(coreEvalBridgeHandler).fromBridge(bridgeMessage);
+  t.log('add-STARS proposal executed');
+
+  t.like(readLatest(`${auctioneerPath}.book1`), {
+    currentPriceLevel: null,
+  });
+});
+
+test('addAsset to active auction', async t => {
+  const {
+    advanceTimeTo,
+    readLatest,
+    // @ts-expect-error override
+    proposal,
+  } = t.context;
+  const { EV } = t.context.runUtils;
+
+  t.like(readLatest(`${auctioneerPath}.book0`), { startPrice: null });
+
+  const auctioneerKit = await EV.vat('bootstrap').consumeItem('auctioneerKit');
+  const schedules = await EV(auctioneerKit.creatorFacet).getSchedule();
+  const { nextAuctionSchedule } = schedules;
+  t.truthy(nextAuctionSchedule);
+  const nextStartTime = nextAuctionSchedule.startTime;
+  const fiveMinutes = harden({
+    relValue: 5n * 60n,
+    timerBrand: nextStartTime.timerBrand,
+  });
+  const futureBusyTime = TimeMath.addAbsRel(nextStartTime, fiveMinutes);
+
+  await advanceTimeTo(futureBusyTime);
+
+  t.log('launching proposal');
+
+  const proposalJSON = JSON.stringify(proposal);
+  const proposalMod = proposalJSON
+    .replaceAll('STARS', 'PLANETS')
+    .replaceAll('ibc/987C17B1', 'ibc/987C17B2');
+  const proposalNew = JSON.parse(proposalMod);
+
+  const bridgeMessage = {
+    type: 'CORE_EVAL',
+    evals: proposalNew.evals,
+  };
+  t.log({ bridgeMessage });
+
+  const coreEvalBridgeHandler = await EV.vat('bootstrap').consumeItem(
+    'coreEvalBridgeHandler',
+  );
+  EV(coreEvalBridgeHandler).fromBridge(bridgeMessage);
+
+  const nextEndTime = nextAuctionSchedule.endTime;
+  const afterEndTime = TimeMath.addAbsRel(nextEndTime, fiveMinutes);
+  await advanceTimeTo(afterEndTime);
+
+  const schedulesAfter = await EV(auctioneerKit.creatorFacet).getSchedule();
+  // brand equality is broken
```

Comment:
how and why?

Reply:
I'm surprised object identity of brands isn't preserved here. I thought maybe you were getting the brands from 2 different places - one from the kernel and one from vstorage - but not so; both are coming from calls to the `creatorFacet` by way of the kernel.

`EV` used to preserve object identity in such circumstances, I believe.

@gibson042 @FUDCo was there a regression in #8373 ? Or is my mental model just off?

Reply:
Well, it's partially that your mental model is a bit off but there _was_ a change that is part of the story here.  You **cannot** directly compare object identity on the stuff you get back from `EV`, though reason is not actually `EV` itself but `kmarshal`, which previously had internal state that we've now gotten rid of because it was fundamentally incompatible with importation into different compartments.  To address this there are now a handful of `matchXXX` functions in the same `supports` module with `EV`.  To compare two object references (such as to brands) for equality, the one you want is `matchRef`.

(Though to be clear, there _are_ some important limitations of `EV` that people seem to be oblivious to that they need to be more broadly educated about, but I don't think they're involved in this particular case.)

Reply:
> You cannot directly compare object identity on the stuff you get back from `EV`

noted. thanks for confirming.

Reply:
I updated the comment to clarify the reason it can't use `TimeMath.compareAbs()`.